api/src/events/mod.rs

```diff
@@ -112,17 +121,20 @@ pub async fn process(
                 let vault =
                     find_vault_id_by_project_id(db.get(), payload.project_id.clone()).await?;
 
-                let (status, signature) = create_raw_transaction(
-                    key.clone(),
-                    payload.transaction.context("transaction not found")?,
-                    payload.project_id.clone(),
-                    vault,
-                    db,
-                    fireblocks,
-                    rpc,
-                    TxType::MintEdition,
-                )
-                .await?;
+                let (status, signature) =
+                    match create_raw_transaction(db, fireblocks, rpc, RawTxParams {
+                        key: key.clone(),
+                        transaction: payload.transaction.context("transaction not found")?,
+                        project_id: payload.project_id.clone(),
+                        vault,
+                        tx_type: TxType::MintEdition,
+                        treasury_vault_id,
+                    })
+                    .await
+                    {
+                        Ok((status, signature)) => (status, signature.to_string()),
+                        Err(_) => (TransactionStatus::FAILED, String::new()),
+                    };
```

Comment:
Also why not have create_raw_transaction return this signature instead of mapping it each time?